rust/operator-binary/src/superset_controller.rs

```diff
@@ -368,3 +504,109 @@ fn build_server_rolegroup_statefulset(
 pub fn error_policy(_error: &Error, _ctx: Context<Ctx>) -> Action {
     Action::requeue(Duration::from_secs(5))
 }
+
+pub fn append_authentication_volumes_and_volume_mounts(
+    authentication_class: &AuthenticationClass,
+    volumes: &mut Vec<Volume>,
+    volume_mounts: &mut Vec<VolumeMount>,
+) {
+    let authentication_class_name = authentication_class.metadata.name.as_ref().unwrap();
+
+    match &authentication_class.spec.provider {
+        AuthenticationClassProvider::Ldap(ldap) => {
+            if let Some(bind_credentials) = &ldap.bind_credentials {
+                let volume_name = format!("{authentication_class_name}-bind-credentials");
+                let volume_mount_path = format!("/secrets/{volume_name}");
+                volumes.push(
+                    VolumeBuilder::new(&volume_name)
+                        .csi(build_secret_operator_volume(
+                            &bind_credentials.secret_class,
+                            &bind_credentials.scope,
+                        ))
+                        .build(),
+                );
+                volume_mounts.push(
+                    VolumeMountBuilder::new(&volume_name, volume_mount_path)
+                        .read_only(true)
+                        .build(),
+                );
+            }
+
+            if let Some(tls) = &ldap.tls {
+                match &tls.verification {
+                    TlsVerification::None {} => {}
+                    TlsVerification::Server(server_verification) => {
+                        match &server_verification.ca_cert {
+                            CaCert::WebPki {} => {}
+                            CaCert::SecretClass(secret_class_nane) => {
+                                append_certificate_secret_class(
+                                    volumes,
+                                    volume_mounts,
+                                    authentication_class_name,
+                                    secret_class_nane,
+                                );
+                            }
+                        }
+                    }
+                    TlsVerification::Mutual(mutual_verification) => {
+                        append_certificate_secret_class(
+                            volumes,
+                            volume_mounts,
+                            authentication_class_name,
+                            &mutual_verification.cert_secret_class,
+                        );
+                    }
+                }
+            }
+        }
+    }
+}
+
+fn append_certificate_secret_class(
+    volumes: &mut Vec<Volume>,
+    volume_mounts: &mut Vec<VolumeMount>,
+    authentication_class_name: &str,
+    secret_class_name: &str,
+) {
+    let volume_name = format!("{authentication_class_name}-tls-certificate");
+    let volume_mount_path = format!("/certificates/{volume_name}");
```

Comment:
Same questions about naming and paths here.

Reply:
`/stackable/certificates/{volume_name}` would be fine?

Reply:
That's fine by me.